### ISSUE TITLE
`panic!`ing twice crashes PG with "Illegal instruction"

### ISSUE THREAD

**Issue by auterium** [bluejekyll/pg-extend-rs#118]
### OS
Ubuntu 18.04 (Windows Subsystem for Linux)

### Code
`panicking` example package

### Issue
Rust `panic!()` is caught and handled properly on the first call, but if called again, the server crashes:
![image](https://user-images.githubusercontent.com/4732632/71627734-9466c880-2bf4-11ea-9534-bafed31b2aef.png)

Here's PG logs:
![image](https://user-images.githubusercontent.com/4732632/71627750-aa748900-2bf4-11ea-9ab4-4548eb310aba.png)

### That little extra weird stuff
 This also happens when using `error!()` macro, but it requires 3 executions:
![image](https://user-images.githubusercontent.com/4732632/71627979-f3790d00-2bf5-11ea-9bca-ee3c4339c6ab.png)

Here;s PG logs:
![image](https://user-images.githubusercontent.com/4732632/71628002-10addb80-2bf6-11ea-8f2f-8c964f8e72b1.png)

**Comment by bluejekyll** [comment]
Error, fatal and panic all cause PG to abort the transaction, where panic will cause a full restart. The illegal obstruction is concerning though.

Is this a double panic? Or is it a panic, and then a different transaction also panicking?

**Comment by auterium** [comment]
Unless `psql` is not setting `AUTOCOMMIT` then yes, they are separate transactions panicking

**Comment by bluejekyll** [comment]
This is happening on Windows? I wonder if this behavior can be reproduced on a vanilla Linux box.

**Comment by auterium** [comment]
Same error happens in a Docker container running PG10
![image](https://user-images.githubusercontent.com/4732632/71630303-7acc7d80-2c02-11ea-93d3-5fb2aea44a92.png)

**Comment by bluejekyll** [comment]
I haven’t looked into this, yet, but what is the expected behavior of Postgres when it panics? It might do exactly this. It would be interesting if we had a test case that showed this with and without pg-extend. That would require a vanilla C extension I think.

**Comment by auterium** [comment]
Looking at PG source code, I found this:
![image](https://user-images.githubusercontent.com/4732632/71676126-ab85f180-2d7f-11ea-8da5-fb4bb82d0220.png)

The important part is:
> If elevel >= ERROR, the call will not return; we try to inform the compiler of that via pg_unreachable()

So, if the log level is `ERROR` or higher, there will be a different behaviour (no return). Perhaps this might explain the weird behaviour?

**Comment by bluejekyll** [comment]
Yes. That’s what I was trying to explain. Error and higher log statements have the side effect of killing the transaction/process/server, depending on which you use.

It’s really ugly imo, but that’s the way it works.
